### ISSUE TITLE
Whiteboard data loss when theme switch[Bug]

### ISSUE THREAD

**Issue by imaryandokania** [ankidroid/Anki-Android#8202]
###### Reproduction Steps

1. The whiteboard draw option
2. when pen is used
3. And then night mode is switched, the whiteboard gets cleared

###### Actual Result
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/61244704/111068790-505a9d80-84f2-11eb-8aad-5ba9d29406d1.gif)

###### Expected Result
Caching of the whiteboard data

###### Debug info
I am ready to work on this , probably can be done by saving data as svg before changing the theme.

Possible changes in restartActivityInvalidateBackstack function 

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by imaryandokania** [comment]
I am working on this and I think I will be able to resolve this

**Comment by mikehardy** [comment]
This is a special case of #3654 - that issue has info on how to easily reproduce

**Comment by imaryandokania** [comment]
I am thinking for an alternative way, this will allow the restart of the app but with all datas intact , the coordinates of the drawing areeas can be added to sql lite

**Comment by imaryandokania** [comment]
@david-allison-1 what are your views on this?

**Comment by david-allison-1** [comment]
My understanding is that we don't lose memory when the theme is switched. The whiteboard is a small quantity of data which shouldn't be enough to require persistent storage via SQLite and associated complexity.

I expect that we can serialise the path or keep it in memory and redraw it given this data once the theme has been changed. There should be code in the codebase which does similar for variables.

**Comment by mikehardy** [comment]
Making sure it works across Activity restarts is good enough, and making sure all Activities work across Activity restarts (can be tested by turning off keep activities in developer settings) is much much more important than trying to save whiteboard data across Application restarts. I'm not sure that's even a feature people would want?

Let's leave that for now and focus on Activity restart correctness - e.g. #3654

**Comment by imaryandokania** [comment]
Yes I agree, we can work as an add-on if user requests it.For now its better to focus more on restart of all activity when theme changes.

**Comment by BrayanDSO** [comment]
Solved with the new whiteboard
